src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/MissingGeneratorAttributeRuleTests.cs

```diff
@@ -0,0 +1,406 @@
+                },
+                FixedState =
+                {
+                    Sources = { fixedCode, SourceGeneratorStub_CSharp },
+                    AdditionalReferences = { AdditionalMetadataReferences.CodeAnalysisReference},
+                },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestHierarchy_VisualBasic()
+        {
+            var code = @"
+Imports Microsoft.CodeAnalysis 
+
+Public MustInherit Class CustomGeneratorBase
+    Implements ISourceGenerator
+
+    Public MustOverride Sub Initialize(context As GeneratorInitializationContext) Implements ISourceGenerator.Initialize
+    Public MustOverride Sub Execute(context As GeneratorExecutionContext) Implements ISourceGenerator.Execute
+End Class
+
+Public Class [|CustomGenerator|]
+    Inherits CustomGeneratorBase
+
+    Public Overrides Sub Initialize(context As GeneratorInitializationContext)
+
+    End Sub
+
+    Public Overrides Sub Execute(context As GeneratorExecutionContext)
+
+    End Sub
+End Class";
+
+            var fixedCode = @"
+Imports Microsoft.CodeAnalysis 
+
+Public MustInherit Class CustomGeneratorBase
+    Implements ISourceGenerator
+
+    Public MustOverride Sub Initialize(context As GeneratorInitializationContext) Implements ISourceGenerator.Initialize
+    Public MustOverride Sub Execute(context As GeneratorExecutionContext) Implements ISourceGenerator.Execute
+End Class
+
+<Generator>
+Public Class CustomGenerator
+    Inherits CustomGeneratorBase
+
+    Public Overrides Sub Initialize(context As GeneratorInitializationContext)
+
+    End Sub
+
+    Public Overrides Sub Execute(context As GeneratorExecutionContext)
+
+    End Sub
+End Class";
+
+            await new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources = { code , SourceGeneratorStub_VisualBasic },
+                    AdditionalReferences = { AdditionalMetadataReferences.CodeAnalysisReference}
+                },
+                FixedState =
+                {
+                    Sources = { fixedCode, SourceGeneratorStub_VisualBasic },
+                    AdditionalReferences = { AdditionalMetadataReferences.CodeAnalysisReference},
+                },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestHierarchy_InheritedAttribute_CSharp()
+        {
+            var code = @"
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[Generator]
+public abstract class CustomGeneratorBase : ISourceGenerator
+{
+    public abstract void Initialize(GeneratorInitializationContext context);
+
+    public abstract void Execute(GeneratorExecutionContext context);
+}
+
+public class CustomGenerator : CustomGeneratorBase
+{
+    public override void Initialize(GeneratorInitializationContext context) {}
+
+    public override void Execute(GeneratorExecutionContext context)
+    {
+    }
+}";
```

Comment:
Can you add another test with `CustomGenerator` being in a new document, and the overrides are not there (just the class)?

This will have compile errors, but what I'm intending to test here is to trigger the analyzer and codefix without `using Microsoft.CodeAnalysis;` being there, to see if the generator will be added as fully-qualified name or the using will get added by the codefix.

Reply:
> to see if the generator will be added as fully-qualified name or the using will get added by the codefix.

I came looking for this, but by the time I reached this reply I realized that if a user fully-qualified `ISourceGenerator` in code, they probably want `GeneratorAttribute` to be fully-qualified as well.

Reply:
Agree on adding test, but I think leaving fully qualified is ideal. Sam did a good job of capturing my initial reasoning about not adding an imports annotation.

Reply:
@sharwell The case I'm referring to involves a document that doesn't contain `ISourceGenerator`, so the attribute will be the first thing from `Microsoft.CodeAnalysis` namespace.